### PR TITLE
Exporting DAOTests root and simple changed in FlashcardDAO

### PR DESCRIPTION
**Pull request by VanDavv**
Tests now extends one abstract class, which contain the common methods for all DAOTests.
FlashcardDAO is changed, so that Flashcard entity now contain deckID

### CODE REVIEW

**Comment by mlotysz** [comment]
First: it should compile. :)

File: backend/app/src/test/java/com/bls/patronage/resources/FlashcardResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @After
+    public void tearDown() {
+        reset(dao);
+    }
+
+    @Test
+    public void updateFlashcard() {
+        final Response response = resources.client().target("/decks/" + flashcard.getDeckID() + "/flashcards/" + flashcard.getId())
```
Comment (mlotysz) [inline backend/app/src/test/java/com/bls/patronage/resources/FlashcardResourceTest.java]: We don't want to concatenate strings here. Check how could it be done in elegant way: https://github.com/blstream/StudyBox_Backend/commit/5c8299b97206c64bd9eba694b62137f840239734

File: backend/db/src/main/java/com/bls/patronage/db/model/Flashcard.java [hunk 2]
```diff
@@ -7,6 +7,13 @@
     private final String answer;
     private final UUID deckID;
 
+    public Flashcard() {
```
Comment (mlotysz) [inline backend/db/src/main/java/com/bls/patronage/db/model/Flashcard.java]: I think we should prohibit create such object. Flashcard without question or answer? Remove this constructor, please.

**Comment by mlotysz** [comment]
Rebase this PR, please - it cannot be merged cleanly anymore.

File: backend/db/src/main/java/com/bls/patronage/db/model/Flashcard.java (hunk 2 above)
Comment (VanDavv) [inline backend/db/src/main/java/com/bls/patronage/db/model/Flashcard.java]: It is needed for Jackson deserialization

File: backend/app/src/test/java/com/bls/patronage/resources/FlashcardResourceTest.java (hunk 1 above)
Comment (VanDavv) [inline backend/app/src/test/java/com/bls/patronage/resources/FlashcardResourceTest.java]: fixed

File: backend/app/src/main/java/com/bls/patronage/api/FlashcardRepresentation.java [hunk 3]
```diff
@@ -18,11 +20,15 @@ public FlashcardRepresentation(String question, String answer) {
     }
 
     @JsonProperty
+    @NotNull
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/api/FlashcardRepresentation.java]: Are those two means: `@NotEmpty`?

**Comment by mlotysz** [comment]
Thanks!

File: backend/app/src/main/java/com/bls/patronage/api/FlashcardRepresentation.java (hunk 3 above)
Comment (VanDavv) [inline backend/app/src/main/java/com/bls/patronage/api/FlashcardRepresentation.java]: It will be changed, together with adding @length annotation

**Pull request merged**